includes/User.php

```diff
@@ -483,18 +494,62 @@ public static function newFromConfirmationCode( $code ) {
 	}
 
 	/**
-	 * Create a new user object using data from session or cookies. If the
-	 * login credentials are invalid, the result is an anonymous user.
+	 * Creates a MediaWiki User object based on the token given in the HTTP request.
 	 *
-	 * @param $request WebRequest object to use; $wgRequest will be used if
-	 *        ommited.
-	 * @return User object
+	 * @param \WebRequest $request the HTTP request data as an object
+	 *
+	 * @return \User on successful authentication
 	 */
-	public static function newFromSession( WebRequest $request = null ) {
-		$user = new User;
-		$user->mFrom = 'session';
-		$user->mRequest = $request;
-		return $user;
+	public static function newFromToken( \WebRequest $request ) {
```

Comment:
It makes sense, thank you for the answer. Based on the conversation, I would add a comment in the method doc stating that this method returns anonymous user in some of the cases.